godnsbl.go

```diff
@@ -97,91 +94,88 @@ func Blacklists() []string {
 	}
 }
 
-// RBLResults holds the results of the lookup.
-type RBLResults struct {
-	// List is the RBL that was searched
-	List string `json:"list"`
-	// Host is the host or IP that was passed (i.e. smtp.gmail.com)
-	Host string `json:"host"`
-	// Results is a slice of Results - one per IP address searched
-	Results []Result `json:"results"`
-}
-
-// Result holds the individual IP lookup results for each RBL search.
+// Result holds the individual result of an IP lookup for an RBL search.
 // nolint:maligned
 type Result struct {
-	// Address is the IP address that was searched
-	Address string `json:"address"`
 	// Listed indicates whether or not the IP was on the RBL
 	Listed bool `json:"listed"`
-	// RBL lists sometimes add extra information as a TXT record
-	// if any info is present, it will be stored here.
+	// RBL lists sometimes add extra information as a TXT record.
+	// If any info is present, it will be stored here.
 	Text string `json:"text"`
-	// Error represents any error that was encountered (DNS timeout, host not
-	// found, etc.) if any
-	Error bool `json:"error"`
-	// ErrorType is the type of error encountered if any
-	ErrorType error `json:"error_type"`
 }
 
 // Reverse the octets of a given IPv4 address
-// 64.233.171.108 becomes 108.171.233.64.
-func Reverse(ip net.IP) string {
-	if ip.To4() == nil {
-		return ""
-	}
-
-	splitAddress := strings.Split(ip.String(), ".")
-
-	for i, j := 0, len(splitAddress)-1; i < len(splitAddress)/2; i, j = i+1, j-1 {
-		splitAddress[i], splitAddress[j] = splitAddress[j], splitAddress[i]
+// '64.233.171.108' becomes '108.171.233.64'.
+// If it's not an v4 IP, returns nil.
+func Reverse(ip net.IP) net.IP {
+	ip4 := ip.To4()
+	if ip4 == nil {
+		return nil
 	}
 
-	return strings.Join(splitAddress, ".")
+	return net.IPv4(ip4[3], ip4[2], ip4[1], ip4[0])
 }
 
-func query(rbl string, host string, r *Result) {
-	r.Listed = false
-
-	lookup := fmt.Sprintf("%s.%s", host, rbl)
+func query(rbl string, ip net.IP) (*Result, error) {
+	lookup := fmt.Sprintf("%s.%s", Reverse(ip).String(), rbl)
 
 	res, err := net.LookupHost(lookup)
+	if err != nil && !recordNotFound(err) { // If the record doesn't exists, the IP isn't blacklisted
```

Comment:
Previously it would just show up as an error if the records didn't exist... I've change it to not show up as an error.
If the record doesn't exists, my understanding is that that means that the IP _is not listed_ in the blacklist...
@toshiya-doi that's correct, isn't it?

Reply:
`NXDOMAIN` code represents that an IP is not listed. I suppose `dnsErr.IsNotFound` means `NXDOMAIN`. 🤔 

> Look up this name in the DNS as a domain name ("A" record). This will return either an address, indicating that the client is listed; or an "NXDOMAIN" ("No such domain") code, indicating that the client is not.
https://en.wikipedia.org/wiki/Domain_Name_System-based_Blackhole_List#DNSBL_queries

Reply:
Yep, pretty sure that's what `dnsErr.IsNotFound` means 👍